### PR TITLE
feat(Tree): add behaviors with list/listitem roles

### PR DESCRIPTION
**Pull request by silviuaavram**
Needed in order to make VO navigation possible. It will read the tree as a list, but it will calculate the list item accordingly, based on the level, setsize, posinset.

Feedback on how to name these behaviors?

Also added missing tests for tree behaviors.

### CODE REVIEW

File: packages/react/src/lib/accessibility/Behaviors/Tree/treeItemAsListItemBehavior.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        ...(props.hasSubtree && {
+          role: 'listitem',
+        }),
+      },
+    },
+  })
+}
+
+export type TreeItemBehaviorProps = {
+  /** Indicated if tree title has a subtree */
```
Comment (lucivpav) [inline packages/react/src/lib/accessibility/Behaviors/Tree/treeItemAsListItemBehavior.ts]:
```suggestion
  /** Indicates whether `TreeTitle` has a subtree */
```

File: packages/react/src/lib/accessibility/Behaviors/Tree/treeItemBehavior.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ const treeItemBehavior: Accessibility<TreeItemBehaviorProps> = props => ({
     root: {
       role: 'none',
       ...(props.hasSubtree && {
-        'aria-expanded': props.open,
+        'aria-expanded': props.open ? 'true' : 'false',
```
Comment (lucivpav) [inline packages/react/src/lib/accessibility/Behaviors/Tree/treeItemBehavior.ts]: There are a few instances in Stardust where we pass a `boolean` to `aria-expanded`. Let's report it as an issue?

File: packages/react/src/lib/accessibility/Behaviors/Tree/treeTitleAsListItemTitleBehavior.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        }),
+      },
+    },
+  })
+}
+
+export default treeTitleAsListItemTitleBehavior
+
+type TreeTitleBehavior = {
+  /** Indicated if tree title has a subtree */
```
Comment (lucivpav) [inline packages/react/src/lib/accessibility/Behaviors/Tree/treeTitleAsListItemTitleBehavior.ts]:
```suggestion
  /** Indicates whether `TreeTitle` has a subtree */
```

File: packages/react/test/specs/behaviors/treeItemAsListItemBehavior-test.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { treeItemAsListItemBehavior } from 'src/lib/accessibility'
+
+describe('TreeItemAsListItemBehavior', () => {
+  describe('role', () => {
+    test(`is 'none' if not a leaf`, () => {
+      const expectedResult = treeItemAsListItemBehavior({ hasSubtree: true })
```
Comment (lucivpav) [inline packages/react/test/specs/behaviors/treeItemAsListItemBehavior-test.tsx:6]: Misleading variable name. This comment applies to other tests as well.
```suggestion
      const behavior = treeItemAsListItemBehavior({ hasSubtree: true })
```

File: packages/react/test/specs/behaviors/treeItemAsListItemBehavior-test.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { treeItemAsListItemBehavior } from 'src/lib/accessibility'
+
+describe('TreeItemAsListItemBehavior', () => {
+  describe('role', () => {
+    test(`is 'none' if not a leaf`, () => {
```
Comment (lucivpav) [inline packages/react/test/specs/behaviors/treeItemAsListItemBehavior-test.tsx]: How about adding a test where its a leaf?

File: packages/react/test/specs/behaviors/treeItemBehavior-test.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    })
+  })
+
+  describe('role', () => {
+    test(`is 'none' if not a leaf`, () => {
+      const expectedResult = treeItemBehavior({ hasSubtree: true })
+      expect(expectedResult.attributes.root.role).toEqual('treeitem')
+    })
+
+    test(`is 'treeitem' if not a leaf`, () => {
```
Comment (lucivpav) [inline packages/react/test/specs/behaviors/treeItemBehavior-test.tsx]:
```suggestion
    test(`is 'treeitem' if a leaf`, () => {
```

File: packages/react/test/specs/behaviors/treeItemAsListItemBehavior-test.tsx (hunk 4 above)
Comment (silviuaavram) [reply packages/react/test/specs/behaviors/treeItemAsListItemBehavior-test.tsx:6]: all the custom behavior tests have the same variable name. I'd leave it like this for now.

File: packages/react/src/lib/accessibility/Behaviors/Tree/treeItemBehavior.ts (hunk 2 above)
Comment (silviuaavram) [reply packages/react/src/lib/accessibility/Behaviors/Tree/treeItemBehavior.ts]: hmm let me take a look

File: packages/react/test/specs/behaviors/treeItemAsListItemBehavior-test.tsx (hunk 4 above)
Comment (silviuavram) [reply packages/react/test/specs/behaviors/treeItemAsListItemBehavior-test.tsx]: for consistency I made this one boolean as well. what do you think?

File: packages/react/src/lib/accessibility/Behaviors/Tree/treeItemBehavior.ts (hunk 2 above)
Comment (silviuaavram) [reply packages/react/src/lib/accessibility/Behaviors/Tree/treeItemBehavior.ts]: for consistency I made this one boolean as well. what do you think?

Comment (lucivpav) [reply packages/react/src/lib/accessibility/Behaviors/Tree/treeItemBehavior.ts]: Looks good to me

File: packages/react/src/lib/accessibility/Behaviors/Tree/treeAsListBehavior.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  return _.merge(behavior, {
+    attributes: {
+      root: {
+        role: 'list',
+      },
+    },
+  })
+}
+
+type TreeBehaviorProps = {} & Pick<AccessibilityAttributes, 'aria-labelledby'>
```
Comment (layershifter) [inline packages/react/src/lib/accessibility/Behaviors/Tree/treeAsListBehavior.ts]:
```suggestion
type TreeBehaviorProps = Pick<AccessibilityAttributes, 'aria-labelledby'>
```

_nit_, currently looks a bit strange

**Review by layershifter — APPROVED** [review]

**Review by jurokapsiar — CHANGES_REQUESTED** [review]
Looks good to me and I think it is as good as it currently can be.

Can you please add an example? It's quite complex to put all the behaviors on the right place

**Review by jurokapsiar — APPROVED** [review]

**Comment by jurokapsiar** [comment]
pls add changelog before merging

**Pull request merged**